src/bootstrap/test.rs

```diff
@@ -1613,55 +1613,6 @@ impl Step for CrateLibrustc {
     }
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
-pub struct CrateNotDefault {
```

Comment:
Should everything be made `pub(crate)` instead of `pub`? That would ensure that dead code like this is detected by rustc.